packages/react-charting/UnitTests/VerticalBarChartUT.test.tsx

```diff
@@ -0,0 +1,241 @@
+        left: 10,
+        right: 10,
+        top: 10,
+        bottom: 10,
+      };
+      const width = 1000;
+      const instance = new VerticalBarChartBase({
+        data: stringPoints,
+        width: width,
+        margins: margin,
+      });
+      expect(instance).toBeDefined();
+      instance._xAxisLabels = stringPoints!.map((point: IVerticalBarChartDataPoint) => point.x as string);
+      instance._getMargins(margin);
+      const margins = instance._getDomainMargins(1000);
+      expect(margins).toBeDefined();
+      expect(margins.left).toEqual(468);
+      expect(margins.right).toEqual(468);
+      expect(margins.top).toEqual(10);
+      expect(margins.bottom).toEqual(10);
+    });
+    test('Should return the correct margins when total width is less than required width', () => {
+      const margin = {
+        left: 10,
+        right: 10,
+        top: 10,
+        bottom: 10,
+      };
+      const width = 1000;
+      const instance = new VerticalBarChartBase({
+        data: stringPoints,
+        width: width,
+        margins: margin,
+      });
+      expect(instance).toBeDefined();
+      instance._getMargins(margin);
+      instance._xAxisLabels = stringPoints!.map((point: IVerticalBarChartDataPoint) => point.x as string);
+      const margins = instance._getDomainMargins();
+      expect(margins).toBeDefined();
+      expect(margins.left).toEqual(18);
+      expect(margins.right).toEqual(18);
+      expect(margins.top).toEqual(10);
+      expect(margins.bottom).toEqual(10);
+    });
+  });
+  runTest('Get scales', () => {
+    it('Should return scales for numeric axis', () => {
+      const margin = {
+        left: 10,
+        right: 10,
+        top: 10,
+        bottom: 10,
+      };
+      const instance = new VerticalBarChartBase({
+        data: chartPoints,
+        theme: DarkTheme,
+        margins: margin,
+      });
+      expect(instance).toBeDefined();
+      instance._adjustProps();
+      instance._yMax = Math.max(
+        d3Max(instance._points, (point: IVerticalBarChartDataPoint) => point.y)!,
+        instance.props.yMaxValue || 0,
+      );
+      instance._xAxisLabels = chartPoints!.map((point: IVerticalBarChartDataPoint) => point.x as string);
+      instance._getMargins(margin!);
+      const containerHeight = 500;
+      const containerWidth = 800;
+      const isNumericAxis = true;
+
+      const scales = instance._getScales(containerHeight, containerWidth, isNumericAxis);
+
+      expect(scales.xBarScale).toBeDefined();
+      expect(scales.yBarScale).toBeDefined();
+      expect(scales.xBarScale(-1000)).toBeLessThan(0);
+      expect(scales.xBarScale(20000)).toBeLessThanOrEqual(containerWidth);
+      expect(scales.xBarScale(40000)).toBeGreaterThan(containerWidth);
+      expect(scales.yBarScale(-5000)).toBeLessThan(0);
+      expect(scales.yBarScale(5000)).toBeLessThanOrEqual(containerHeight);
+      expect(scales.yBarScale(60000)).toBeGreaterThan(containerHeight);
```

Comment:
add an equality check also

Reply:
Here and other tests below.

Reply:
equality assert is still missing.

Reply:
for StringAxis, the value for which xBarScale is equal to containerWidth cannot be determined. For all others, I have added the equality cases too.